src/shared/udev-util.c

```diff
@@ -192,7 +228,17 @@ int device_wait_for_initialization(sd_device *device, const char *subsystem, use
 
         /* Check again, maybe things changed. Udev will re-read the db if the device wasn't initialized
          * yet. */
-        if (sd_device_get_is_initialized(device) > 0) {
+        if (!device && devlink) {
+                r = stat(devlink, &st);
+                if (r < 0 && errno != ENOENT)
+                        return log_error_errno(errno, "Failed to stat() %s: %m", devlink);
+                if (r == 0) {
+                        r = sd_device_new_from_devnum(&device, 'b', st.st_rdev);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to initialize device from %s: %m", devlink);
+                }
+        }
```

Comment:
given that we basically do the same thing twice here, maybe add a local helper device_new_from_dev_path() or so, that does this

Reply:
Done